sips/sip-clarity4/sip-clarity4.md

```diff
@@ -0,0 +1,378 @@
+# Preamble
+
+**SIP Number:** XYZ
+
+**Title:** Clarity Smart Contract Language, version 4
+
+**Authors:**
+
+- Adriano Di Luzio <adriano@bitcoinl2labs.com>
+- Brice Dobry <brice@hiro.so>
+- Marvin Janssen <marvin@ryder.id>
+- Jude Nelson <jude@stacks.org>
+
+**Consideration:** Technical
+
+**Type:** Consensus
+
+**Status:** Draft
+
+**Created:** 2025-06-20
+
+**License:** CC0-1.0
+
+**Sign-off:**
+
+**Discussions-To:** https://github.com/stacksgov/sips
+
+# Abstract
+
+This SIP details Version 4 of the Clarity Smart Contract Language, which
+introduces new operations to make it easier to build secure Smart Contracts on
+Stacks.
+
+# License and Copyright
+
+This SIP is made available under the terms of the CC0 (Creative Commons Zero)
+license, available at https://opensource.org/licenses/CC0-1.0. This SIP’s
+copyright is held by the Stacks Open Internet Foundation.
+
+# Introduction
+
+The Clarity smart contract language powers decentralized applications across a
+wide range of domains. Over years of real-world use, developers have encountered
+several pain points when interacting with other, sometimes untrusted, contracts.
+Securely calling unknown contracts unlocks powerful use cases, but demands
+careful safeguards.
+
+This SIP addresses common feedback and requests from builders in the ecosystem.
```

Comment:
```suggestion
This SIP addresses common feedback and requests from developers in the ecosystem.
```
nit - this is the only use of `builders`. the rest of the document refers to `developers` -  recommended change for consistency.